docs/upgrade.md

```diff
@@ -0,0 +1,149 @@
+# Upgrading KubeVirt
+
+This document outlines the step by step process of upgrading kubevirt
+from the perspective of the kubevirt-operator.  These steps can also
+be done by hand.
+
+**Major Upgrade** - Upgrade from 1.0 to 2.0
+**Minor Upgrade** - Upgrade from 1.0 to 1.1
+
+### Major Upgrade
+*kubevirt-1.1 to kubevirt-2.0*
+
+*When the kubevirt-operator sees the Virt CR change from version 1.1 to 2.0 it will...*
+ - Rollout kubevirt-2.0 virt-handler daemonsets
+ - Test virt-handler functionality with kubevirt-2.0 data
+ - Scale down virt-controllers pods
+ - Scale down virt-api pods
+ - Register kubevirt-2.0 CRD webhooks
+ - Update to kubevirt-2.0 CRDs
+ - Rollout kubevirt-2.0 virt-controllers
+ - Test virt-controller functionality with kubevirt-2.0 data
+ - Rollout kubevirt-2.0 virt-api
+ - Test virt-api functionality with kubevirt-2.0 data
+
+
+### Minor Upgrade
+*kubevirt-1.1 to kubevirt-1.2*
```

Comment:
I think these instructions are different than below.  Maybe we should just remove?

Reply:
I don't think we should remove it. 
It is an overall plan, we may want to use it in the future.

Reply:
You have the minor upgrade plan below.  Downstream can replace version numbers in the docs so we don't need to have two sections titled minor upgrade.